src/Health/HealthController.ts

```diff
@@ -0,0 +1,205 @@
+import BitcoinCore = require('bitcoin-core')
+import { inject, injectable } from 'inversify'
+import { Collection, Db } from 'mongodb'
+import * as Pino from 'pino'
+
+import { childWithFileName } from 'Helpers/Logging'
+import { IPFSHashFailure } from 'Interfaces'
+import { IPFS } from './IPFS'
+
+@injectable()
+export class HealthController {
+  private readonly db: Db
+  private readonly collection: Collection
+  private readonly bitcoinCore: BitcoinCore
+  private readonly logger: Pino.Logger
+  private readonly ipfs: IPFS
+
+  constructor(
+    @inject('Logger') logger: Pino.Logger,
+    @inject('DB') db: Db,
+    @inject('BitcoinCore') bitcoinCore: BitcoinCore,
+    @inject('IPFS') ipfs: IPFS
+  ) {
+    this.logger = childWithFileName(logger, __filename)
+    this.db = db
+    this.collection = this.db.collection('health')
+    this.bitcoinCore = bitcoinCore
+    this.ipfs = ipfs
+  }
+
+  async checkMongoConnection(): Promise<any> {
+    this.logger.trace('MongoHealth', 'Checking Mongo...')
+    const connection = await this.db.stats()
+    const mongoIsConnected = connection.ok === 1 ? true : false
+    await this.collection.updateOne(
+      { name: 'mongoConnected' },
+      {
+        $set: {
+          mongoIsConnected,
+        },
+      },
+      { upsert: true }
+    )
+  }
+
+  async checkIpfsConnection(): Promise<any> {
+    this.logger.trace('ipfsHealth', 'Checking IPFS...')
```

Comment:
You can leave the first argument out here, changing it to `this.logger.trace('Checking IPFS...')`.

This will log the following:

```
[2018-10-10T00:31:07.250Z] TRACE (31451 on Laumint): Checking IPFS...
    module: "Health"
    file: "HealthController"
```

The `module` and `file` properties that `this.logger` already provide enough info in this case IMO.

If you'd like to add the method name like we're doing in other places in the code base, please pass an object as the first parameter, so it gets merged (instead of the string getting concatenated), same way you're doing it in `getBlockchainInfo`.